server/src/main/java/org/elasticsearch/transport/TransportService.java

```diff
@@ -966,19 +977,12 @@ public boolean isForceExecution() {
                         }
 
                         @Override
-                        public void onFailure(Exception e) {
-                            handleSendToLocalException(channel, e, action);
-                        }
+                        public void onFailure(Exception e) {}
```

Comment:
Same here, can't do this. We need to use the callback when using the exeuctor with an `AbstractRunnable`